beacon.go

```diff
@@ -133,20 +147,26 @@ func validateServerConfig() error {
 	return nil
 }
 
-func parseInput(r *http.Request) (string, string, int64, error) {
+type queryParams struct {
```

Comment:
Optional, but here you might actually want a 'query' type.

type query struct {
   refName string
   allele string
   ...
}

func (q *query) Execute(bq *bigquery.Client) (bool, error) {
   ...
}

sort of thing

Reply:
This worked out great. I've refactored the code to split the functionality at it's more readable now.